src/toolbox/setup/windows.ts

```diff
@@ -1,5 +1,190 @@
-import { print } from 'gluegun'
+import { 
+  print,
+  filesystem, 
+  system 
+} from 'gluegun'
+import {
+  INSTALL_PATH,
+  INSTALL_DIR,
+  EXPORTS_FILE_PATH,
+  MODDABLE_REPO
+} from './constants'
+import upsert from '../patching/upsert'
+
+const regedit = require("regedit")
+
+interface Result {
+  success: boolean,
+  info?: string
+}
+
+enum RegType {
+  REG_SZ = "REG_SZ",
+  REG_EXPAND_SZ = "REG_EXPAND_SZ",
+  REG_DWORD = "REG_DWORD",
+  REG_QWORD = "REG_QWORD",
+  REG_MULTI_SZ = "REG_MULTI_SZ",
+  REG_DEFAULT = "REG_DEFAULT"
+}
+
+async function setEnv(name: string, value: string, type: RegType): Promise<Result> {
+  let result
+  try {
+    result = await regedit.promisified.list([`HKCU\\ENVIRONMENT`])
+  } catch (error) {
+    return {success: false, info: "Error while reading user environment variables from registry"}
+  }
+  
+  try {
+    if (result) {
+      let reg = {
+        "HKCU\\ENVIRONMENT": {}
+      };
+      (reg as any)["HKCU\\ENVIRONMENT"][name] = {
+        value,
+        type
+      }
+      await regedit.promisified.putValue(reg)
+      return {success: true}
+    }
+  } catch (error) {
+    return {success: false, info: "Error while saving environment variable to registry"}
+  }
+  return {success: false}
+}
```

Comment:
I don't think we need a `Result` type since we're using Promises, so we can lean into the `try...catch` control flow around the result of this async function. It also appears the `result` value from reading the environment variables is unused, so we can remove that logic from the top of the function. The `reg` variable is also updated to use the `RegistryItemPutCollection` from `regedit`. 


```suggestion
async function setEnv(name: string, value: RegistryItemValue): Promise<void> {
  try {
    const reg: RegistryItemPutCollection = {
      'HKCU\\ENVIRONMENT': {
        [name]: value,
      },
    }
    await regedit.putValue(reg)
  } catch (error) {
    throw new Error('Error while saving environment variable to registry')
  }
}
```